src/ardupilot/ArduPilotInterface.cpp

```diff
@@ -0,0 +1,21 @@
+#include "ArduPilotProtocol.h"
+
+using namespace robot::types;
+using namespace navtypes;
+using namespace std::chrono_literals;
+
+namespace {
+net::ardupilot::ArduPilotProtocol ardupilot_protocol;
+} // namespace
+
+namespace gps {
+DataPoint<gpscoords_t> readGPSCoords() {
+	return ardupilot_protocol.getGPS();
+}
+} // namespace gps
+
+namespace robot {
+DataPoint<double> readIMUHeading() {
+	return ardupilot_protocol.getIMU().getData().yaw;
```

Comment:
We should check if the datapoint is valid before calling `getData()`, this will fail otherwise

Reply:
What do we want to return in the case that it is not valid?